### ISSUE TITLE
Attempting to extend workspace lifetime too long has unhelpful generic error message

### ISSUE THREAD

**Issue by ketang** [coder/coder#2812]
This is too generic:
![image](https://user-images.githubusercontent.com/33678/177059072-c8a98e99-463c-46ba-9460-a10d4d2a6748.png)

The API does return useful information that is not displayed:
```
{"message":"Bad extend workspace request.","validations":[{"field":"deadline","detail":"new deadline is greater than template allows"}]}
```

**Comment by bpmct** [comment]
@presleyp I remember seeing another note about generic error messages on the frontend. Thoughts on consolidating these?

**Comment by presleyp** [comment]
@bpmct Yes, the epic I'm in the middle of fleshing out will address this. #1943

**Comment by ketang** [comment]
Has #1943 addressed this?
